services/web/server/src/simcore_service_webserver/conversations/_socketio.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+import datetime
+from typing import Final
+
+from aiohttp import web
+from models_library.conversations import (
+    ConversationID,
+    ConversationMessageGetDB,
+    ConversationMessageID,
+)
+from models_library.projects import ProjectID
+from models_library.socketio import SocketMessageDict
+from models_library.users import UserID
+from pydantic import AliasGenerator, BaseModel, ConfigDict
+from pydantic.alias_generators import to_camel
+
+from ..socketio.messages import send_message_to_standard_group
+
+SOCKET_IO_CONVERSATION_MESSAGE_CREATED_EVENT: Final[str] = (
+    "conversation:message:created"
+)
+SOCKET_IO_CONVERSATION_MESSAGE_DELETED_EVENT: Final[str] = (
+    "conversation:message:deleted"
+)
+SOCKET_IO_CONVERSATION_MESSAGE_UPDATED_EVENT: Final[str] = (
+    "conversation:message:updated"
+)
+
+
+class BaseConversationMessage(BaseModel):
+    conversation_id: ConversationID
+    message_id: ConversationMessageID
+
+    model_config = ConfigDict(
+        populate_by_name=True,
+        from_attributes=True,
+        alias_generator=AliasGenerator(
+            serialization_alias=to_camel,
+        ),
+    )
+
+
+class ConversationMessageCreated(BaseConversationMessage):
+    content: str
+    created: datetime.datetime
+
+
+class ConversationMessageUpdated(BaseConversationMessage):
+    content: str
+    modified: datetime.datetime
+
+
+class ConversationMessageDeleted(BaseConversationMessage): ...
+
+
+async def _send_message_to_recipients(app, recipients, notification_message):
+    await asyncio.gather(
+        *[
+            send_message_to_standard_group(app, recipient, notification_message)
+            for recipient in recipients
+        ]
+    )
```

Comment:
for info. gather is unbounded. this creates the coroutine and then runs them all concurrently. 
we do have `limited_gather` that allows to limit the number of calls, or use `limited_as_completed` to also not pre-create all the coroutines but on the go using a coroutine generator.
```python
async for _ in limited_as_completed((send_message_to_standard_group(app, recipient, notification_message)
            for recipient in recipients)):
  ...
```

Reply:
Thanks for the tip